File: emergency-centre-typescript/src/toxic-gas-sensor.ts
import Action from "./action";
import { GasType } from "./gas-type";
import Sensor from "./sensor";
import { Vendor } from "./vendor";

export default class ToxicGasSensor extends Sensor {
  private readonly concentration: number;
  private readonly gasType: GasType;

  constructor(
    id: number,
    vendor: Vendor,
    actions: Action[],
    concentration: number,
    gasType: GasType
  ) {
    super(id, vendor, actions);
    this.concentration = concentration;
    this.gasType = gasType;
  }

  getConcentration(): number {
    return this.concentration;
  }

  getGasType(): GasType {
    return this.gasType;
  }

  override toString(): string {
    return `
    ToxicGasSensor{id=${this.getId()}, vendor=${this.getVendor()}, 
    concentration=${this.concentration}, gasType=${this.gasType}}
    `;
  }
}

File: emergency-centre-c#/EmergencyCentre/ToxicGasSensor.cs
using System;
using EmergencyCentreDotNet.EmergencyCentre.Action;

namespace EmergencyCentreDotNet.EmergencyCentre;

public class ToxicGasSensor(int id, Vendor vendor, List<SensorAction> actions, double concentration, GasType gasType) : Sensor(id, vendor, actions)
{
    public double Concentration { get; private set; } = concentration;
    public GasType GasType { get; private set; } = gasType;

    public override string ToString()
    {
        return base.ToString() + $"_(concentration: {Concentration}, gasType: {GasType})";
    }
}
